File: app/src/main/java/com/u1fukui/bbs/repository/ThreadRepository.java
package com.u1fukui.bbs.repository;


import android.os.SystemClock;

import com.u1fukui.bbs.model.ApiResponse;
import com.u1fukui.bbs.model.Comment;
import com.u1fukui.bbs.model.User;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import io.reactivex.Single;

public class ThreadRepository {

    public Single<List<Comment>> fetchCommentList(final long threadId) {
        return Single.create(e -> {
            SystemClock.sleep(1000);

            List<Comment> list = new ArrayList<>();
            for (int i = 1; i <= 20; i++) {
                User author = new User(i, "コメンター" + i);
                Comment comment = new Comment(i, threadId, i, "コメント", author, i, false, new Date());
                list.add(comment);
            }
            e.onSuccess(list);
        });
    }

    public Single<ApiResponse> postThread() {
        return Single.create(e -> {
            SystemClock.sleep(1000);
            e.onSuccess(new ApiResponse());
        });
    }

    public Single<ApiResponse> postComment() {
        return Single.create(e -> {
            SystemClock.sleep(1000);
            e.onSuccess(new ApiResponse());
        });
    }
}

File: app/src/main/java/com/u1fukui/bbs/repository/ThreadRepository.kt
package com.u1fukui.bbs.repository


import android.os.SystemClock
import com.u1fukui.bbs.api.ThreadApi
import com.u1fukui.bbs.model.ApiResponse
import com.u1fukui.bbs.model.Comment
import com.u1fukui.bbs.model.EmptyResponse
import io.reactivex.Single
import kotlinx.coroutines.Deferred

class ThreadRepository(private val threadApi: ThreadApi) {

    fun fetchCommentList(
        threadId: Long,
        lastId: Long?,
        pageSize: Int
    ): Deferred<List<Comment>> =
            threadApi.fetchCommentList(threadId, lastId, pageSize)

    fun postThread(): Single<ApiResponse> {
        return Single.create { e ->
            SystemClock.sleep(1000)
            e.onSuccess(EmptyResponse(200))
        }
    }

    fun postComment(): Single<ApiResponse> {
        return Single.create { e ->
            SystemClock.sleep(1000)
            e.onSuccess(EmptyResponse(200))
        }
    }
}
